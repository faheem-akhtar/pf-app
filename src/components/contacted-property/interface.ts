import { ContactedPropertyTypeEnum } from 'enums/contacted-property/type.enum';

export interface ContactedPropertyInterface {
  /**
   * ID
   */
  id: string;

  /**
   * The associated property id
   */
  propertyId: number;

  /**
   * Contact type
   */
  contactType: ContactedPropertyTypeEnum;

  /**
   * The contacted date
   */
  contactDate: string;
}
