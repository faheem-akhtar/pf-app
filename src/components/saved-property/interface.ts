export interface SavedPropertyInterface {
  /**
   * ID
   */
  id: string;

  /**
   * The associated property id
   */
  propertyId: number;

  /**
   * The saved property date
   */
  saveDate: string;
}
