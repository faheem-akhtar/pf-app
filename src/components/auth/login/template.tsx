import React, { FunctionComponent } from 'react';

import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { ErrorMessageComponent } from 'components/error-message/component';
import { IconFacebookTemplate } from 'components/icon/facebook-template';
import { IconGoogleTemplate } from 'components/icon/google-template';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { InputBaseComponent } from 'library/input/base/component';

import styles from '../auth.module.scss';
import { AuthLoginTemplatePropsInterface } from './template-props.interface';

export const AuthLoginTemplate: FunctionComponent<AuthLoginTemplatePropsInterface> = (props) => (
  <div data-testid='AuthLoginComponent'>
    <h1 className={styles.heading}>{props.t('sign-in')}</h1>
    {/* Facebook */}
    <ButtonTemplate
      type='button'
      className={domClassMerge(styles.button, styles.facebook)}
      componentType={ButtonComponentTypeEnum.tertiary}
      size={ButtonSizeEnum.small}
      onClick={props.onFacebookLoginClick}
      icon={{ component: IconFacebookTemplate, position: ButtonIconPositionEnum.left }}
    >
      {props.t('auth/sign-in-facebook')}
    </ButtonTemplate>
    {/* Google */}
    <ButtonTemplate
      type='button'
      className={domClassMerge(styles.button, styles.google)}
      componentType={ButtonComponentTypeEnum.tertiary}
      size={ButtonSizeEnum.small}
      onClick={props.onGoogleLoginClick}
      icon={{ component: IconGoogleTemplate, position: ButtonIconPositionEnum.left }}
    >
      {props.t('auth/sign-in-google')}
    </ButtonTemplate>
    <div className={styles.divider}>
      <span className={styles.text}>{props.t('auth/or')}</span>
    </div>
    <AuthLoaderComponent isEnabled={props.isLoading} isCentered={true} />
    {props.errorMessage && <ErrorMessageComponent message={props.errorMessage} />}
    <form onSubmit={props.onFormSubmit}>
      <div className={styles['input-area']}>
        <InputBaseComponent
          id='email'
          name='email'
          placeholder={props.t('email')}
          type='email'
          value={props.email}
          error={!!props.errors[AuthLoginFieldEnum.email]}
          onChange={props.onEmailChange}
          errorText={props.errors[AuthLoginFieldEnum.email]}
        />
      </div>
      <div className={styles['input-area']}>
        <InputBaseComponent
          id='password'
          name='password'
          placeholder={props.t('password')}
          type='password'
          value={props.password}
          error={!!props.errors[AuthLoginFieldEnum.password]}
          onChange={props.onPasswordChange}
          errorText={props.errors[AuthLoginFieldEnum.password]}
        />
      </div>
      <div className={styles['input-area']} onClick={props.onForgotPassword}>
        <div className={domClassMerge(styles.link, styles['forgot-password'])}>{props.t('auth/forgot-password')}?</div>
      </div>
      <div className={styles['input-area']}>
        <ButtonTemplate
          className={styles.button}
          type='submit'
          componentType={ButtonComponentTypeEnum.primary}
          size={ButtonSizeEnum.regular}
        >
          {props.t('log-in')}
        </ButtonTemplate>
      </div>
      <ReCaptchaComponent />
    </form>
    <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
      {props.t('auth/not-registered-yet')}?&nbsp;{props.t('auth/create-account')}
    </div>
  </div>
);
