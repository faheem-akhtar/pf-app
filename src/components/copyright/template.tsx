import { useTranslation } from 'helpers/translation/hook';

export const CopyrightTemplate: React.FunctionComponent = () => {
  const { t } = useTranslation();
  return <div>{t('copyright')}</div>;
};
