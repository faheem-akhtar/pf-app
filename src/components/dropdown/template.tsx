import { ChangeEvent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import { DropdownTemplatePropsInterface } from './template-props.interface';
import { FiltersValueFieldChoiceInterface } from 'components/filters/value/field/choice.interface';

import styles from './dropdown.module.scss';
import { useTranslation } from 'next-i18next';

// TODO-FE[TPNX-3056] remove this file
export const DropdownTemplate = <T extends unknown>({
  title,
  value,
  choices,
  onChange,
  className,
}: DropdownTemplatePropsInterface<T>): JSX.Element => {
  const { t } = useTranslation();
  return (
    <div className={domClassMerge(className, styles.container)}>
      {title && <span className={styles.title}>{title}</span>}
      <select
        value={String(choices.indexOf(choices.find((c) => c.value === value) as FiltersValueFieldChoiceInterface<T>))}
        onChange={(e): void =>
          onChange(choices[parseInt((e as unknown as ChangeEvent<HTMLSelectElement>).target.value, 10)].value)
        }
      >
        {choices.map((choice, index) => (
          <option key={index} value={index}>
            {choice.label ? choice.label : t('Any')}
          </option>
        ))}
      </select>
    </div>
  );
};
