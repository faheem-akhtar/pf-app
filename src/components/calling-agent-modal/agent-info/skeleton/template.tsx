import { SkeletonTemplate } from 'library/skeleton/template';

import styles from './calling-agent-modal-agent-info-skeleton.module.scss';

export const CallingAgentModalAgentInfoSkeletonTemplate = (): JSX.Element => (
  <div className={styles.container}>
    <SkeletonTemplate width='7.04rem' height='7.04rem' class={styles['skeleton--image']} />

    <div className={styles.content}>
      <SkeletonTemplate width='8rem' height='2.4rem' class={styles.skeleton} />
      <SkeletonTemplate width='100%' height='2.5rem' class={styles.skeleton} />
    </div>
  </div>
);
