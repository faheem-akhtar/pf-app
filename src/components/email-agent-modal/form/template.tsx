import { configEmailAgentMustAcceptTerms } from 'config/email-agent/must-accept-terms';
import { configEmailAgentReceiveAdsEnabled } from 'config/email-agent/receive-ads-enabled';

import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { EmailAgentModalFormTemplatePropsInterface } from './template-props.interface';
import { EmailAgentModalFormWidgetAcceptTermsComponent } from './widget/accept-terms-component';
import { EmailAgentModalFormWidgetEmailAlertComponent } from './widget/email-alert-component';
import { EmailAgentModalFormWidgetEmailComponent } from './widget/email-component';
import { EmailAgentModalFormWidgetMessageComponent } from './widget/message-component';
import { EmailAgentModalFormWidgetNameComponent } from './widget/name-component';
import { EmailAgentModalFormWidgetPhoneComponent } from './widget/phone-component';
import { EmailAgentModalFormWidgetReceiveAdvertisingComponent } from './widget/receive-advertising-component';

import styles from './email-agent-modal-form.module.scss';

export const EmailAgentModalFormTemplate = ({
  fieldsValue,
  setFieldsValue,
  onSubmit,
  t,
}: EmailAgentModalFormTemplatePropsInterface): JSX.Element => {
  return (
    <form className={styles.form} onSubmit={onSubmit}>
      <EmailAgentModalFormWidgetNameComponent
        t={t}
        value={fieldsValue.name}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, name: newValue })}
      />
      <EmailAgentModalFormWidgetEmailComponent
        t={t}
        value={fieldsValue.email}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, email: newValue })}
      />
      <EmailAgentModalFormWidgetPhoneComponent
        t={t}
        value={fieldsValue.phone}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, phone: newValue })}
      />
      <EmailAgentModalFormWidgetMessageComponent
        t={t}
        value={fieldsValue.message}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, message: newValue })}
      />
      <EmailAgentModalFormWidgetEmailAlertComponent
        t={t}
        value={fieldsValue.emailAlert}
        onChange={(newValue): void => setFieldsValue({ ...fieldsValue, emailAlert: newValue })}
      />
      {configEmailAgentMustAcceptTerms && (
        <EmailAgentModalFormWidgetAcceptTermsComponent
          t={t}
          value={fieldsValue.acceptTerms}
          onChange={(newValue): void => setFieldsValue({ ...fieldsValue, acceptTerms: newValue })}
        />
      )}
      {configEmailAgentReceiveAdsEnabled && (
        <EmailAgentModalFormWidgetReceiveAdvertisingComponent
          t={t}
          value={fieldsValue.receiveAdvertising}
          onChange={(newValue): void => setFieldsValue({ ...fieldsValue, receiveAdvertising: newValue })}
        />
      )}

      <ButtonTemplate
        type='submit'
        size={ButtonSizeEnum.regular}
        componentType={ButtonComponentTypeEnum.primary}
        className={styles.button}
      >
        {t('agent-modal/cta-send-message')}
      </ButtonTemplate>
    </form>
  );
};
