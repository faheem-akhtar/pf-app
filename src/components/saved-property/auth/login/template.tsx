import React, { FunctionComponent } from 'react';

import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginTemplatePropsInterface } from 'components/auth/login/template-props.interface';
import { ErrorMessageComponent } from 'components/error-message/component';
import { IconFacebookTemplate } from 'components/icon/facebook-template';
import { IconGoogleTemplate } from 'components/icon/google-template';
import { IconThinCheckmarkTemplate } from 'components/icon/thin/checkmark-template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import styles from './saved-property-auth-login.module.scss';

export const SavedPropertyAuthLoginTemplate: FunctionComponent<AuthLoginTemplatePropsInterface> = (props) => (
  <div data-testid='saved-property-auth-login-template'>
    <h1 className={styles.heading}>{props.t('saved_property/auth-login-title')}</h1>
    <ul className={styles.list}>
      <li>
        <IconThinCheckmarkTemplate class={styles.list__icon} /> {props.t('saved_property/auth-login-line-1')}
      </li>
      <li>
        <IconThinCheckmarkTemplate class={styles.list__icon} /> {props.t('saved_property/auth-login-line-2')}
      </li>
      <li>
        <IconThinCheckmarkTemplate class={styles.list__icon} /> {props.t('saved_property/auth-login-line-3')}
      </li>
    </ul>
    <ButtonTemplate
      className={styles.button}
      type='button'
      componentType={ButtonComponentTypeEnum.primary}
      size={ButtonSizeEnum.regular}
      onClick={props.onLogin}
    >
      {props.t('log-in')}
    </ButtonTemplate>
    <div className={styles.divider}>
      <span className={styles.text}>{props.t('auth/or')}</span>
    </div>
    <AuthLoaderComponent isEnabled={props.isLoading} isCentered={true} />
    {props.errorMessage && <ErrorMessageComponent message={props.errorMessage} />}
    {/* Google */}
    <ButtonTemplate
      type='button'
      className={domClassMerge(styles.button, styles.google)}
      componentType={ButtonComponentTypeEnum.tertiary}
      size={ButtonSizeEnum.small}
      onClick={props.onGoogleLoginClick}
      icon={{ component: IconGoogleTemplate, position: ButtonIconPositionEnum.left }}
    >
      {props.t('auth/sign-in-google')}
    </ButtonTemplate>
    {/* Facebook */}
    <ButtonTemplate
      type='button'
      className={domClassMerge(styles.button, styles.facebook)}
      componentType={ButtonComponentTypeEnum.tertiary}
      size={ButtonSizeEnum.small}
      onClick={props.onFacebookLoginClick}
      icon={{ component: IconFacebookTemplate, position: ButtonIconPositionEnum.left }}
    >
      {props.t('auth/sign-in-facebook')}
    </ButtonTemplate>
  </div>
);
