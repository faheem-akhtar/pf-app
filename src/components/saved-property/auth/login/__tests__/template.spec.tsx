/**
 * @jest-environment jsdom
 */
import { render, RenderResult } from '@testing-library/react';

import { AuthLoginTemplatePropsInterface } from 'components/auth/login/template-props.interface';

import { SavedPropertyAuthLoginTemplate } from '../template';

describe('SavedPropertyAuthLoginTemplate', () => {
  let renderResult: RenderResult;
  let props: AuthLoginTemplatePropsInterface;

  beforeEach(() => {
    props = {
      isLoading: false,
      errorMessage: '',
      email: '',
      password: '',
      errors: {},
      t: jest.fn(),
      onEmailChange: jest.fn(),
      onPasswordChange: jest.fn(),
      onForgotPassword: jest.fn(),
      onLogin: jest.fn(),
      onRegister: jest.fn(),
      onFacebookLoginClick: jest.fn(),
      onGoogleLoginClick: jest.fn(),
      onFormSubmit: jest.fn(),
    };

    renderResult = render(<SavedPropertyAuthLoginTemplate {...props} />);
  });

  it('renders without throwing any errors', () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it('should render the error message', () => {
    const { getByText } = render(<SavedPropertyAuthLoginTemplate {...props} errorMessage='This is an error message' />);

    expect(getByText('This is an error message')).toBeTruthy();
  });
});
