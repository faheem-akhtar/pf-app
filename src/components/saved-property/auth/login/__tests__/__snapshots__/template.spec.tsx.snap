// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedPropertyAuthLoginTemplate renders without throwing any errors 1`] = `
<div>
  <div
    data-testid="saved-property-auth-login-template"
  >
    <h1
      class="heading"
    />
    <ul
      class="list"
    >
      <li>
        <svg
          class="list__icon"
          viewBox="4 5 17 13"
        >
          <path
            clip-rule="evenodd"
            d="M4.5 12.5993L8.96969 17.6139C9.45785 18.1287 10.2493 18.1287 10.7375 17.6139L20.5 7.60728C20.6953 7.40136 20.6953 7.06749 20.5 6.86157C20.3047 6.65564 19.9882 6.65564 19.7929 6.86157L10.0303 16.8682C9.93272 16.9711 9.77443 16.9711 9.6768 16.8682L5.20711 11.8535C5.01184 11.6476 4.69526 11.6476 4.5 11.8535C4.30474 12.0595 4.30474 12.3933 4.5 12.5993Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
         
      </li>
      <li>
        <svg
          class="list__icon"
          viewBox="4 5 17 13"
        >
          <path
            clip-rule="evenodd"
            d="M4.5 12.5993L8.96969 17.6139C9.45785 18.1287 10.2493 18.1287 10.7375 17.6139L20.5 7.60728C20.6953 7.40136 20.6953 7.06749 20.5 6.86157C20.3047 6.65564 19.9882 6.65564 19.7929 6.86157L10.0303 16.8682C9.93272 16.9711 9.77443 16.9711 9.6768 16.8682L5.20711 11.8535C5.01184 11.6476 4.69526 11.6476 4.5 11.8535C4.30474 12.0595 4.30474 12.3933 4.5 12.5993Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
         
      </li>
      <li>
        <svg
          class="list__icon"
          viewBox="4 5 17 13"
        >
          <path
            clip-rule="evenodd"
            d="M4.5 12.5993L8.96969 17.6139C9.45785 18.1287 10.2493 18.1287 10.7375 17.6139L20.5 7.60728C20.6953 7.40136 20.6953 7.06749 20.5 6.86157C20.3047 6.65564 19.9882 6.65564 19.7929 6.86157L10.0303 16.8682C9.93272 16.9711 9.77443 16.9711 9.6768 16.8682L5.20711 11.8535C5.01184 11.6476 4.69526 11.6476 4.5 11.8535C4.30474 12.0595 4.30474 12.3933 4.5 12.5993Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
         
      </li>
    </ul>
    <button
      class="button primary regular button"
    />
    <div
      class="divider"
    >
      <span
        class="text"
      />
    </div>
    
    <button
      class="button tertiary small button google"
    >
      <svg
        class="icon icon--left"
        viewBox="0 0 22 22"
      >
        <g>
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              transform="translate(-183.000000, -256.000000)"
            >
              <g
                transform="translate(183.000000, 256.000000)"
              >
                <g>
                  <path
                    d="M21.56,11.25 C21.56,10.47 21.49,9.72 21.36,9 L11,9 L11,13.255 L16.92,13.255 C16.665,14.63 15.89,15.795 14.725,16.575 L14.725,19.335 L18.28,19.335 C20.36,17.42 21.56,14.6 21.56,11.25 L21.56,11.25 Z"
                    fill="#4285F4"
                  />
                  <path
                    d="M11,22 C13.97,22 16.46,21.015 18.28,19.335 L14.725,16.575 C13.74,17.235 12.48,17.625 11,17.625 C8.135,17.625 5.71,15.69 4.845,13.09 L1.17,13.09 L1.17,15.94 C2.98,19.535 6.7,22 11,22 L11,22 Z"
                    fill="#34A853"
                  />
                  <path
                    d="M4.845,13.09 C4.625,12.43 4.5,11.725 4.5,11 C4.5,10.275 4.625,9.57 4.845,8.91 L4.845,6.06 L1.17,6.06 C0.425,7.545 0,9.225 0,11 C0,12.775 0.425,14.455 1.17,15.94 L4.845,13.09 L4.845,13.09 Z"
                    fill="#FBBC05"
                  />
                  <path
                    d="M11,4.375 C12.615,4.375 14.065,4.93 15.205,6.02 L18.36,2.865 C16.455,1.09 13.965,0 11,0 C6.7,0 2.98,2.465 1.17,6.06 L4.845,8.91 C5.71,6.31 8.135,4.375 11,4.375 L11,4.375 Z"
                    fill="#EA4335"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    </button>
    <button
      class="button tertiary small button facebook"
    >
      <svg
        class="icon icon--left"
        viewBox="0 0 12 23"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#FEFEFE"
            transform="translate(-190.000000, -198.000000)"
          >
            <g
              transform="translate(166.000000, 186.000000)"
            >
              <g>
                <path
                  d="M31.9328743,35 L27.6347873,35 L27.6347873,23.5309121 L24.7675854,23.5309121 L24.7675854,19.5782401 L27.6347873,19.5768961 L27.6301953,17.248481 C27.6301953,14.0238238 28.5045409,12.0618241 32.3028177,12.0618241 L35.4653137,12.0618241 L35.4653137,16.0151681 L33.4888657,16.0151681 C32.0096518,16.0151681 31.9385864,16.5675073 31.9385864,17.5984881 L31.9326503,19.5768961 L35.4874341,19.5768961 L35.0684897,23.529568 L31.9357303,23.5309121 L31.9328743,35 Z"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </button>
  </div>
</div>
`;
